static/src/js/containers/MapContainer/MapContainer.tsx

```diff
@@ -361,52 +367,43 @@ export const MapContainer: React.FC<MapContainerProps> = (props) => {
     setStartDrawing(false)
   }, [setStartDrawing])
 
-  const { tags } = focusedCollectionMetadata
-  const [gibsTag] = getValueForTag('gibs', tags) || []
-
-  // Get the colormap data for the currently focused collection
-  const colorMapState: ColormapMetadata = useMemo(() => {
-    let colorMapData = {}
-
-    // If the collection has a GIBS tag and the GIBS layer is available for the current projection, use the colormap data
-    if (gibsTag && hasGibsLayerForProjection(gibsTag, projection)) {
-      const { product } = gibsTag
-      colorMapData = colormapsMetadata[product] || {}
-    }
-
-    return colorMapData
-  }, [gibsTag, colormapsMetadata, projection])
+  // Helper function to get GIBS tags available for the current projection
+  const getLayersForProjection = useCallback(() => {
+    if (!mapLayers) return []
 
-  const { colorMapData: colorMap = {} } = colorMapState
+    return mapLayers.filter((tag) => hasGibsLayerForProjection(tag, projection))
+  }, [mapLayers, projection])
 
+  // Extract the actual colormap data from the state
   // Get GIBS data to pass to the map within each granule
-  let gibsData: Partial<GibsData> = {}
-  if (gibsTag) {
-    const {
-      antarctic_resolution: antarcticResolution,
-      arctic_resolution: arcticResolution,
-      format,
-      geographic_resolution: geographicResolution,
-      layerPeriod,
-      product
-    } = gibsTag
-
-    let resolution
-    if (projection === projectionCodes.antarctic) {
-      resolution = antarcticResolution
-    } else if (projection === projectionCodes.arctic) {
-      resolution = arcticResolution
-    } else {
-      resolution = geographicResolution
+  const layersForProjection = getLayersForProjection()
+
+  const imageryLayers: ImageryLayers = useMemo(() => {
+    const imageryLayersObject: ImageryLayers = {
+      layerData: [],
+      toggleLayerVisibility,
+      setMapLayersOrder,
+      setLayerOpacity
     }
-
-    gibsData = {
-      format,
-      layerPeriod,
-      product,
-      resolution
+    // If the collection has a GIBS tag and the GIBS layer is available for the current projection, use the colormap data
+    if (layersForProjection.length > 0) {
+      // Get colormap data for all available GIBS tags
+      layersForProjection.forEach((layer) => {
+        const { product } = layer
+        const productColormap = colormapsMetadata[product]
+
+        if (productColormap && productColormap.colorMapData) {
+          // Store colormap data by product name
+          imageryLayersObject.layerData.push({
+            ...layer,
+            colormap: productColormap.colorMapData
+          } as ImageryLayerItem)
+        }
+      })
     }
-  }
+
+    return imageryLayersObject
+  }, [mapLayers, colormapsMetadata, projection, getLayersForProjection])
```

Comment:
You are iterating over `layersForProjection` but it isn't in the dependency array. `mapLayers` is not used at all

```suggestion
  }, [colormapsMetadata, layersForProjection])
```